### PR TITLE
Adds wanted posters, printable from security record computers.

### PR DESCRIPTION
**Pull request by Cruix**
4910882a-c02e-11e5-9acf-c9d3321d957c

### CODE REVIEW

**Comment by Bawhoppen** [comment]
see people, your irrational hatred of yog is unfounded

**Comment by Supermichael777** [comment]
can it not switch between black and white?

**Comment by Bawhoppen** [comment]
just a sprite thing we can change later easily

**Comment by WJohn** [comment]
Yeah the sprite is a major :-1: from me. I think it's pretty obvious why.

**Comment by ike709** [comment]
:+1: except for the sprite

**Comment by coiax** [comment]
Can't we just have the sprite generated from their stored security photograph?

**Comment by Jordie0608** [comment]
I'm pretty sure it is; the sprite complaints are regarding the actual poster itself

**Comment by pudl** [comment]
The images of players is a pain, just make it sort of like the yellowish wild west posters and on examine is says who it is and shows a small version of the players sprite like when you examine any other object

**Comment by PKPenguin321** [comment]
@pudl that would be exceedingly lame compared to this

**Comment by KorPhaeron** [comment]
This could use a better sprite but this is one of the more commonly requested features

**Comment by KorPhaeron** [comment]
Also this is literally the first feature Cruix has added to our codebase and has several fixes in the past, I think we could probably let this slide

**Comment by Cruix** [comment]
@Bawhoppen you lied to me, you said the freeze was over.

**Comment by Bawhoppen** [comment]
theres a permanent rule of no PRs without a fix nowadays, the all encompassing freeze is over though

File: code/game/objects/effects/wanted_poster.dm [hunk 1]
```diff
@@ -0,0 +1,56 @@
+/obj/item/weapon/poster/legit/wanted
+	var/poster_desc
+	icon_state = "rolled_poster"
+
+/obj/item/weapon/poster/legit/wanted/New(turf/loc, icon/person_icon, wanted_name, description)
+	..(loc)
+	name = "wanted poster ([wanted_name])"
+	desc = "A wanted poster for [wanted_name]."
+	poster_desc = description
+	qdel(resulting_poster)
```
Comment (optimumtact) [inline code/game/objects/effects/wanted_poster.dm]: where is this var set?

File: code/game/objects/effects/wanted_poster.dm [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+/obj/structure/sign/poster/wanted/New(var/icon/person_icon, var/person_name, var/description)
+	if(!person_icon)
+		qdel(src)
+		return
+	name = "wanted poster ([person_name])"
+	wanted_name = person_name
+	desc = description
+
+	var/obj/item/weapon/photo/photo = new /obj/item/weapon/photo()
+	photo.photocreate(null, icon(person_icon, dir = SOUTH))//copy the image so we don't mess with the one in the record.
```
Comment (optimumtact) [inline code/game/objects/effects/wanted_poster.dm]: why not person_icon = icon(person_icon, dir = SOUTH)

why are you messing around with a photo obj? What benefit does it add?

File: code/game/objects/effects/wanted_poster.dm [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+	person_icon.Shift(SOUTH, 7)
+	person_icon.Crop(7,4,26,30)
+	person_icon.Crop(-5,-2,26,29)
+	the_icon.Blend(person_icon, ICON_OVERLAY)
+	the_icon.Blend(icon_foreground, ICON_OVERLAY)
+
+	the_icon.Insert(the_icon, "wanted")
+	the_icon.Insert(icon('icons/obj/contraband.dmi', "poster_being_set"), "poster_being_set")
+	the_icon.Insert(icon('icons/obj/contraband.dmi', "poster_ripped"), "poster_ripped")
+	icon = the_icon
```
Comment (optimumtact) [inline code/game/objects/effects/wanted_poster.dm]: this is minor, but you should investigate caching a wanted poster so you don't reblend it each time they print a copy of the poster, you could store by users ckey or some other uniq method

File: code/game/objects/effects/wanted_poster.dm (hunk 1 above)
Comment (Cruix) [inline code/game/objects/effects/wanted_poster.dm]: in /obj/item/weapon/poster/New()

File: code/game/objects/effects/wanted_poster.dm (hunk 2 above)
Comment (Cruix) [inline code/game/objects/effects/wanted_poster.dm]: Absolutely nothing, I am removing it now.

File: code/game/objects/effects/wanted_poster.dm (hunk 3 above)
Comment (Cruix) [inline code/game/objects/effects/wanted_poster.dm]: Definitely a good thought, but I am not going to deal with that right now and have a chance of horribly breaking it.

**Comment by KorPhaeron** [comment]
No changelog

**Pull request merged**